src/main/cpp/src/hive_hash.cu

```diff
@@ -419,12 +452,12 @@ class hive_device_row_hasher {
             top.update_cur_hash(single_level_list_hash);
```

Comment:
If I am not wrong, these two lines

            top.update_cur_hash(single_level_list_hash);
            if (--stack_size > 0) { col_stack[stack_size - 1].update_cur_hash(top.get_hash()); }
``` equals to 
```c++
            if (--stack_size > 0) { col_stack[stack_size - 1].update_cur_hash(single_level_list_hash); }

Reply:
That's not correct. 
If the calculation is for a single-layer `list`, removing `top.update_cur_hash(single_level_list_hash);` will result in `return col_stack[0].get_hash();` not giving the correct answer.